### PR TITLE
chore: oracle cosmetic updates

### PR DESCRIPTION
**Pull request by robert-zaremba**
## Description

Various updates when doing oracle review prior to 3.0 release.

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] added appropriate labels to the PR
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Review by adamewozniak — COMMENTED** [review]
some initial comments

File: x/oracle/types/ballot.go [hunk 1]
```diff
@@ -169,16 +168,11 @@ func ClaimMapToSlice(claims map[string]Claim) []Claim {
...
-			Weight:    claim.Weight,
-			WinCount:  claim.WinCount,
-			Recipient: claim.Recipient,
-		}
+		c[i] = claim // makes copy
 		i++
 	}
 	sort.Slice(c, func(i, j int) bool {
-		return c[i].Recipient.String() < c[j].Recipient.String()
+		return bytes.Compare(c[i].Validator, c[j].Validator) < 0
```
[the remark this answers is not in the document]
Comment (zarazan) [reply x/oracle/types/ballot.go:175]: I agree with Woz on this one but it's personal preference.

**Review by zarazan — COMMENTED** [review]
Good cleanup 👍🏻

File: x/oracle/types/ballot.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply x/oracle/types/ballot.go:175]: Comparing bytes by converting to string is usually wrong and has performance hit. Bytes should be compared using slice operator, which is implemented by `bytes.Compare`.

File: x/oracle/types/vote.go [hunk 2]
```diff
@@ -16,8 +17,8 @@ func NewAggregateExchangeRatePrevote(
 	submitBlock uint64,
 ) AggregateExchangeRatePrevote {
 	return AggregateExchangeRatePrevote{
-		Hash:        hash.String(),
-		Voter:       voter.String(),
+		Hash:        hash.String(),  // we could store bytes here!
+		Voter:       voter.String(), // we could store bytes here!
```
Comment (zarazan) [inline x/oracle/types/vote.go:21]: Why not just convert these over to using bytes instead of leaving the comments?

**Review by zarazan — APPROVED** [review]
👍🏻

File: x/oracle/types/ballot.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (robert-zaremba) [reply x/oracle/types/ballot.go:175]: Comment to a documentation about `bytes.Compare` ? This is a std lib function. Doesn't it self explanatory? I don't think we should explain here a popular std lib function.

**Pull request merged**